javaslang-benchmark/src/test/java/javaslang/JmhRunner.java

```diff
@@ -112,4 +114,28 @@ public static void runAndReport(Array<Class<?>> groups, int warmupIterations, in
     public static int aggregate(int x, int y) {
         return x ^ y;
     }
+
+    /** speeds up construction of immutable objects by caching them */
+    private static Map<Tuple2<Integer, Function1<?, ?>>, Object> FUNCTION_CACHE = HashMap.empty();
+
+    @SuppressWarnings("unchecked")
+    public static <T extends Collection<?>, R> R create(Function1<T, R> function, T source, Function1<R, Boolean> assertion) {
+        return create(function, source, source.size(), assertion);
+    }
+
+    @SuppressWarnings("unchecked")
+    public static <T, R> R create(Function1<T, R> function, T source, int elementCount, Function1<R, Boolean> assertion) {
+        final Tuple2<Integer, Function1<?, ?>> key = Tuple.of(elementCount, function);
```

Comment:
Thank you for the thorough review, I appreciate it :).
I tested it and it seems to be working, probably the identity hash is enough in this case (make sense, since I'm usually referencing existing methods here).

If you are against caching, I will remove it, though pcollecections and fjava constructions are really slow for big values :/